### PR TITLE
use ByteSink and ByteSource instead of OutputSupplier and InputSupplier

### PR DESCRIPTION
**Pull request by infynyxx**
They are being deprecated and will eventually be removed in Guava 18.0

### CODE REVIEW

**Comment by drcrallen** [comment]
:+1:

File: extensions/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsDataSegmentPusher.java [hunk 1]
```diff
@@ -93,10 +93,8 @@ private DataSegment createDescriptorFile(DataSegment segment, Path outDir, final
   {
     final Path descriptorFile = new Path(outDir, "descriptor.json");
     log.info("Creating descriptor file at[%s]", descriptorFile);
-    ByteStreams.copy(
-        ByteStreams.newInputStreamSupplier(jsonMapper.writeValueAsBytes(segment)),
-        new HdfsOutputStreamSupplier(fs, descriptorFile)
-    );
+    final ByteSource byteSource = ByteSource.wrap(jsonMapper.writeValueAsBytes(segment));
```
Comment (xvrl) [inline extensions/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsDataSegmentPusher.java]: This is not a requirement, but you can simply chain the methods here to make it more readable

```java
ByteSource
    .wrap(jsonMapper.writeValueAsBytes(segment))
    .copyTo(new HdfsOutputStreamSupplier(fs, descriptorFile));
```

Comment (infynyxx) [inline extensions/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsDataSegmentPusher.java]: Fixed via 5b06ec2

**Comment by xvrl** [comment]
@infynyxx thanks, do you mind squashing commits, and I'll merge

**Comment by infynyxx** [comment]
@xvrl done

**Comment by xvrl** [comment]
@infynyxx tests failed for unrelated reason, merging.

**Pull request merged**